controllers/todo.controller.js

```diff
@@ -15,12 +15,30 @@ module.exports = server => {
     });
 
     server.post('/todos', (req, res) => {
-        const todo = new Todo(todoId++, req.body.name, req.body.done);
+        if (!req.body.name)
+            return res.send(new errs.BadRequestError("No name for new item supplied."));
+
+        if (typeof req.body.name !== 'string')
+            return res.send(new errs.BadRequestError("name should be a string."));
+        if (req.body.done != undefined && typeof req.body.done !== 'boolean')
+            return res.send(new errs.BadRequestError("done should be a boolean."));
+
+        const todo = new Todo(todoId++, req.body.name, req.body.done != undefined ? req.body.done : false);
```

Comment:
`req.body.done != undefined ? req.body.done : false` can be simplified to `req.body.done || false`